Automattic/sensei
# Add Quick Links to Sensei Home REST API.

Fixes #5741 

### Changes proposed in this Pull Request
* Created `Sensei_Home_Quick_Links_Provider` which provides the Quick Links categories and items.
* Created `Sensei_REST_API_Home_Controller_Mapper` which maps from Sensei Home domain to REST API structure.
* Added unit tests for all new classes.
* Added a unit test mocking Provider and Mapper in the Controller.

### Testing instructions
Don't know how worthy it is to test the REST API right now but:
- Login to your local site.
- Copy as CURL any request to the `/wp-json` endpoint.
- Modify the CURL to point to this endpoint: `/wp-json/sensei-internal/v1/home`.
- Check that the output structure matches the one defined in #5676 
- Check that the output content matches the [Sensei Home i2 design](p6rkRX-4so-p2).

### Screenshot / Video
```
$ curl 'https://test2.local/wp-json/sensei-internal/v1/home' ...
{
  "tasks_list": {
    "tasks": [ ... ]
  },
  "quick_links": [
    {
      "title": "Courses",
      "items": [
        {
          "title": "Create a Course",
          "url": "https://test2.local/wp-admin/post-new.php?post_type=course"
        },
        {
          "title": "Import a Course",
          "url": "https://test2.local/wp-admin/edit.php?post_type=course&page=sensei-tools&tool=import-content"
        },
        {
          "title": "Reports",
          "url": "https://test2.local/wp-admin/edit.php?post_type=course&page=sensei_reports"
        }
      ]
    },
    {
      "title": "Settings",
      "items": [
        {
          "title": "Email notifications",
          "url": "https://test2.local/wp-admin/edit.php?post_type=course&page=sensei-settings#email-notification-settings"
        },
        {
          "title": "Learning Mode",
          "url": "https://test2.local/wp-admin/edit.php?post_type=course&page=sensei-settings#course-settings"
        },
        {
          "title": "WooCommerce",
          "url": "https://test2.local/wp-admin/edit.php?post_type=course&page=sensei-settings#woocommerce-settings"
        },
        {
          "title": "Content Drip",
          "url": "https://test2.local/wp-admin/edit.php?post_type=course&page=sensei-settings#sensei-content-drip-settings"
        }
      ]
    },
    {
      "title": "Advanced Features",
      "items": [
        {
          "title": "Interactive Blocks",
          "url": "https://senseilms.com/interactive-blocks"
        },
        {
          "title": "Groups & Cohorts",
          "url": "https://senseilms.com/groups-cohorts"
        },
        {
          "title": "Quizzes",
          "url": "https://senseilms.com/quizzes"
        },
        {
          "title": "Integrations",
          "url": "https://senseilms.com/sensei-lms-integrations/"
        }
      ]
    }
  ],
  "help": [ ... ],
  "guides": { ... },
  "news": { ... },
  "extensions": [ ... ],
  "show_sensei_pro_promo": false,
  "notifications": [ ... ],
  "products_pending_activation": [ ... ]
}
```
